R/http-requests.R

```diff
@@ -65,3 +65,54 @@ gsheets_GET <- function(url) {
   req$content <- XML::xmlToList(req$content)
   req
 }
+
+
+
+#' Create POST request
+#'
+#' Make POST request to Google Sheets/Drive API.
+#'
+#' @param url URL for POST request
+#' @param the_body body of POST request
+gsheets_POST <- function(url, the_body)
+{
+  token <- get_google_token()
+  
+  if(is.null(token)) {
+    stop("Must be authorized in order to perform request")
+  } else {
+    
+    # first look at the url to determine contents, 
+    # must be either talking to "drive" or "spreadsheets" API
+    if(stringr::str_detect(stringr::fixed(url), "drive")) {
+      content_type <- "application/json" # send json to drive api
+    } else {
+      content_type <- "application/atom+xml" # send xml to sheets api
+    } 
+    
+    req <- httr::POST(url, gsheets_auth(token), 
+                      httr::add_headers("Content-Type" = content_type),
```

Comment:
Which argument of `httr::POST()` is the content type being input as? I'd prefer to see explicitly `arg = value` style here. I only feel comfortable with matching on position for first argument and *maybe* the second.